commons/src/main/java/org/infinispan/commons/util/concurrent/jdk8backported/BoundedEquivalentConcurrentHashMapV8.java

```diff
@@ -1140,7 +1140,6 @@ void demoteLowestLIR() {
                   // Non resident was already evicted and now it is no longer in the
                   // queue or the stack so it is effectively gone - however we want to
                   // remove the now null node
-                  removedLIR.setState(Recency.EVICTING);
```

Comment:
Unfortunately this fix is not quite correct.  This will still cause an additional eviction event to be generated.  You can see it by adding an additional line to the test to verify the correct number of events are generated.

We instead need to lock the segment and the node and remove the element if it is still HIR_NONRESIDENT (someone else could have updated) and not generate an event.  The event was generated when it went from HIR_RESIDENT to HIR_NONRESIDENT.